File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/SimpleViewManager.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.uimanager;

import android.view.View;

/**
 * Common base class for most of the {@link ViewManager}s. It provides support for most common
 * properties through extending {@link BaseViewManager}. It also reduces boilerplate by specifying
 * the type of shadow node to be {@link ReactShadowNode} and providing default, empty implementation
 * for some of the methods of {@link ViewManager} interface.
 *
 * @param <T> the view handled by this manager
 */
public abstract class SimpleViewManager<T extends View>
    extends BaseViewManager<T, LayoutShadowNode> {

  @Override
  public LayoutShadowNode createShadowNodeInstance() {
    return new LayoutShadowNode();
  }

  @Override
  public Class<LayoutShadowNode> getShadowNodeClass() {
    return LayoutShadowNode.class;
  }

  @Override
  public void updateExtraData(T root, Object extraData) {}
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/SimpleViewManager.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.uimanager

import android.view.View

/**
 * Common base class for most of the [ViewManager]s. It provides support for most common properties
 * through extending [BaseViewManager]. It also reduces boilerplate by specifying the type of shadow
 * node to be [ReactShadowNode] and providing default, empty implementation for some of the methods
 * of [ViewManager] interface.
 *
 * @param <T> the view handled by this manager
 */
public abstract class SimpleViewManager<T : View> : BaseViewManager<T, LayoutShadowNode>() {

  public override fun createShadowNodeInstance(): LayoutShadowNode {
    return LayoutShadowNode()
  }

  public override fun getShadowNodeClass(): Class<LayoutShadowNode> {
    return LayoutShadowNode::class.java
  }

  public override fun updateExtraData(root: T, extraData: Any?): Unit = Unit
}
